casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -7,79 +7,77 @@ import com.google.protobuf.ByteString
 import coop.rchain.casper.protocol._
 import coop.rchain.casper.util.ProtoUtil
 import coop.rchain.casper.util.rholang.RuntimeManager.StateHash
-import coop.rchain.catscontrib.TaskContrib._
 import coop.rchain.catscontrib._
-import coop.rchain.crypto.codec.Base16
 import coop.rchain.crypto.hash.Blake2b512Random
 import coop.rchain.models.Expr.ExprInstance.GString
 import coop.rchain.models._
 import coop.rchain.rholang.interpreter.accounting.{Cost, CostAccount}
 import coop.rchain.rholang.interpreter.storage.StoragePrinter
 import coop.rchain.rholang.interpreter.{accounting, ChargingReducer, ErrorLog, Runtime}
-import coop.rchain.rspace.internal.{Datum, WaitingContinuation}
-import coop.rchain.rspace.trace.Produce
+import coop.rchain.rspace.internal.Datum
 import coop.rchain.rspace.{Blake2b256Hash, ReplayException}
 import monix.eval.Task
 import monix.execution.Scheduler
 import coop.rchain.catscontrib.TaskContrib._
 
-import scala.annotation.tailrec
 import scala.collection.immutable
-import scala.concurrent.SyncVar
-import scala.concurrent.duration._
-import scala.util.{Failure, Success, Try}
+
+class RuntimeManager private (emptyStateHash: ByteString, runtimeContainer: MVar[Task, Runtime])
+    extends AbstractRuntimeManager[Task](emptyStateHash, runtimeContainer)
```

Comment:
I see you've chosen the path of fixing the F-type per class. I'd prefer to have each method parameterized with just the needed effects, but won't push on that.

Reply:
It's less code to go this way (IMO).